bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpThingHandler.java

```diff
@@ -245,26 +249,23 @@ private void sendHttpValue(String commandUrl, String command) {
     }
 
     private String concatenateUrlParts(String... urlParts) {
-        return Arrays.asList(urlParts).stream().filter(Objects::nonNull).map(String::trim).filter(s -> !s.isEmpty())
+        return Arrays.stream(urlParts).filter(Objects::nonNull).map(String::trim).filter(s -> !s.isEmpty())
                 .map(s -> s.replaceAll("^/|/$", "")).collect(Collectors.joining("/"));
     }
 
-    public ItemValueConverter createConverter(ItemConverterFactory factory, ChannelUID channelUID,
-            HttpChannelConfig channelConfig) {
-        String stateUrl = concatenateUrlParts(config.baseURL, channelConfig.stateExtension);
-        String commandUrl = channelConfig.commandExtension == null ? stateUrl
-                : concatenateUrlParts(config.baseURL, channelConfig.commandExtension);
-
+    private ItemValueConverter createTransformingConverter(AbstractTransformingItemConverter.Factory factory,
+            String commandUrl, ChannelUID channelUID, HttpChannelConfig channelConfig) {
         return factory.create(state -> updateState(channelUID, state), command -> postCommand(channelUID, command),
                 command -> sendHttpValue(commandUrl, command),
                 valueTransformationProvider.getValueTransformation(channelConfig.stateTransformation),
                 valueTransformationProvider.getValueTransformation(channelConfig.commandTransformation), channelConfig);
     }
 
-    @FunctionalInterface
-    private interface ItemConverterFactory {
-        ItemValueConverter create(Consumer<State> updateState, Consumer<Command> postCommand,
-                @Nullable Consumer<String> sendHttpValue, ValueTransformation stateTransformations,
-                ValueTransformation commandTransformations, HttpChannelConfig channelConfig);
+    private ItemValueConverter createGenericConverter(GenericItemConverter.Factory factory, String commandUrl,
+            ChannelUID channelUID, HttpChannelConfig channelConfig, Class<? extends State> clazz) {
+        return factory.create(clazz, state -> updateState(channelUID, state),
+                command -> postCommand(channelUID, command), command -> sendHttpValue(commandUrl, command),
+                valueTransformationProvider.getValueTransformation(channelConfig.stateTransformation),
+                valueTransformationProvider.getValueTransformation(channelConfig.commandTransformation), channelConfig);
```

Comment:
death to duplicate code.
```suggestion
    private <T extends State> ItemValueConverter createGenericConverter(String commandUrl,
            ChannelUID channelUID, HttpChannelConfig channelConfig, Class<T extends State> clazz) {
        AbstractTransformingItemConverter.Factory factory = (state, command, value, stateTrans, commandTrans, config) -> new GenericItemConverter<T>(clazz, state, command, value, stateTrans, commandTrans, config);
        return createTransformingConverter(factory, commandUrl, channelUID, channelConfig);
```